src/request.go

```diff
@@ -578,50 +628,97 @@ func (r *Requests) sendToInflux() {
 						return
 					}
 				}
-			case p := <-r.Output:
+			case p := <-r.Output: // Read a point from r.Output and add it to `points`.
 				if p != nil {
 					points = append(points, *p)
-					p_len = len(points)
-					if p_len == r.Config.limit {
-						log.Info("Batch: Sending to influx ", p_len, " points")
+					pointsLength = len(points)
+
+					// Send to influx if buffer limit is reached.
+					if pointsLength == r.Config.limit {
+						log.Info("Batch: Sending to influx ", pointsLength, " points")
 						if i.sendToInflux(points, 1) {
 							points = []influx.Point{}
 						} else {
-							return
+							log.Warn("Batch: Sending points failed")
+							return // If sending fails, exit the function.
 						}
 					}
-					index++
 				} else {
-					p_len = len(points)
-					if p_len > 0 {
-						log.Info("Batch: Sending to influx ", p_len, " points")
-						if i.sendToInflux(points, 1) {
-							points = []influx.Point{}
+					// p == nil! Send buffered points to Influx and quit.
+					pointsLength = len(points)
+					if pointsLength > 0 {
+						log.Info("Batch: Sending to influx ", pointsLength, " points")
+						if !i.sendToInflux(points, 1) { // Send all points.
+							log.Warn("Batch: Sending remaining points failed")
+						}
+					}
+
+					// If we have any restore dates, this is a restore operation
+					// and we need to call some of the InfluxDB continuous query
+					// statements manually for the data to be aggregated into
+					// the required InfluxDB measurements, since the CQs
+					// configured in InfluxDB will never do that for data of a
+					// certain age.
+					for _, date := range r.Config.restoreDates {
+						log.Info("Running continuous queries for restored data...")
+
+						cqs := []string{
+							`SELECT distinct("uid") AS "uid" INTO "byUid_24h" FROM "telemetry"  WHERE time >= '%s' AND time < '%s' + 1d GROUP BY time(24h),uid,status,install_version `,
+							`SELECT distinct("uid") AS "uid" INTO "by_country_24h" FROM "telemetry"  WHERE time >= '%s' AND time < '%s' + 1d GROUP BY time(24h),country,country_isocode,uid,status `,
+							`SELECT last("container_total") AS "containers", last("project_pod_total") AS "pods" INTO "pods_containers_24h" FROM "telemetry"  WHERE time >= '%s' AND time < '%s' + 1d GROUP BY time(24h),uid,status `,
+							`SELECT last("orch_kubernetes") AS "kubernetes", last("orch_mesos") AS "mesos", last("orch_swarm") AS "swarm", last("orch_windows") AS "windows", last("orch_cattle") AS "cattle" INTO "orchestrators_24h" FROM "telemetry" WHERE record_version = '1'  AND time >= '%s' AND time < '%s' + 1d GROUP BY time(24h),uid,status`,
```

Comment:
I've re-checked, since it's been a while. We only need a part of the statements in `influx-cq.sql` and they need to be adapted to work properly and only do the aggregation for the newly inserted (or replaced) data.